src/connection.toit

```diff
@@ -132,20 +160,25 @@ class Connection:
 
   read_response -> Response:
     if current_reader_: throw "Previous response not yet finished"
-    version := reader_.read_string (reader_.index_of_or_throw ' ')
-    reader_.skip 1
-    status_code := int.parse (reader_.read_string (reader_.index_of_or_throw ' '))
-    reader_.skip 1
-    status_message := reader_.read_string (reader_.index_of_or_throw '\r')
-    reader_.skip 1
-    if reader_.read_byte != '\n': throw "FORMAT_ERROR"
+    try:
+      version := reader_.read_string (reader_.index_of_or_throw ' ')
+      reader_.skip 1
+      status_code := int.parse (reader_.read_string (reader_.index_of_or_throw ' '))
+      reader_.skip 1
+      status_message := reader_.read_string (reader_.index_of_or_throw '\r')
+      reader_.skip 1
+      if reader_.read_byte != '\n': throw "FORMAT_ERROR"
 
-    headers := read_headers_
-    body_reader := body_reader_ headers --request=false
+      headers := read_headers_
 
-    current_reader_ = body_reader
+      body_reader := body_reader_ headers --request=false
 
-    return Response this version status_code status_message headers body_reader
+      current_reader_ = body_reader
+
```

Comment:
Nitty nit: Here the empty line is after assigning to current_reader_. Above it is before. Maybe make this consistent? What kind of grouping are you aiming for here?